plugins/org.eclipse.emfcloud.modelserver.glsp.integration/src/org/eclipse/emfcloud/modelserver/glsp/operations/handlers/EMSBasicOperationHandler.java

```diff
@@ -49,12 +49,10 @@ public EMSBasicOperationHandler() {
    @Override
    public void executeOperation(final T operation) {
       if (handles(operation)) {
-         EMSModelServerAccess modelServerAccess = getModelServerAccess(gModelState);
+         EMSModelServerAccess modelServerAccess = EMSModelState.getModelServerAccess(gModelState);
          executeOperation(operationType.cast(operation), modelServerAccessType.cast(modelServerAccess));
       }
    }
 
-   protected EMSModelServerAccess getModelServerAccess(final GModelState gModelState) {
-      return EMSModelState.getModelServerAccess(gModelState);
-   }
+   protected EMSModelState getEMSModelState() { return EMSModelState.getModelState(gModelState); }
```

Comment:
can't we directly return an EMSNotationModelState here? or is there a case where the gModelState is not an EMSNotationModelState?

Reply:
The `EMS*ModelState` is bound as `GModelState` in the `DiagramModule`. The `GModelState` is used by the base framework. If we would directly inject `EMS*ModelState`, adopters might get troubles if they bind their own ModelState implementation. But if an adopter binds its own ModelState implementation it can inject it then directly of course.
Therefore I think this is a valid workaround.

Reply:
I see your point, but don't we cast it anyway in the operations? And would a user not have to provide a ModelState which extends the EMSNotationModelState if he wants to use our operations?